crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -162,25 +162,20 @@ impl EventCacheStore for SqliteEventCacheStore {
         let conn = self.acquire().await?;
         let data = conn
             .with_transaction::<_, rusqlite::Error, _>(move |txn| {
-                let Some(media) = txn
-                    .query_row::<Vec<u8>, _, _>(
-                        "SELECT data FROM media WHERE uri = ? AND format = ?",
-                        (&uri, &format),
-                        |row| row.get(0),
-                    )
-                    .optional()?
-                else {
-                    return Ok(None);
-                };
-
                 // Update the last access.
+                // We need to do this first so the transaction is in write mode right away.
```

Comment:
I think it's worst mentionning https://sqlite.org/lang_transaction.html#read_transactions_versus_write_transactions here, what do you think?

Reply:
Sure

Reply:
Added.